pkg/validation/oneof.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"github.com/go-openapi/spec"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+	"strings"
+)
+
+// +k8s:openapi-gen=validator(type(OneOf))
+type OneOf struct {
+	fields      []*field.Path
+	common      *field.Path
+	allNamesCsv string
+}
+
+var _ ValidationType = Len{}
+
+func (l OneOf) Init(op OperationType, fields ...*FieldMeta) field.ErrorList {
+	ret := field.ErrorList{}
+	if len(fields) > 0 {
+		for _, v := range fields {
+			if v.Type != "string" {
```

Comment:
Yeah, this is sort of crippled.  If we spent the time to make optional fields into pointers (as they should be), then you could pass a slice of bool, and the generated code would just test the nil-ness of each pointer when prepping.